lib/puppet/provider/aix_object.rb

```diff
@@ -346,7 +346,7 @@ def addcmd(attributes)
   end
 
   def deletecmd
-    [self.class.command(:delete)] + ia_module_args + [@resource[:name]]
+    [self.class.command(:delete), '-p'] + ia_module_args + [@resource[:name]]
```

Comment:
this code is common for `aix` `group` provider and `user` providers and it will add the `-p` argument also to the group delete command